telepresence/runner/runner.py

```diff
@@ -664,12 +664,23 @@ def wait_for_exit(self, main_process: Popen) -> None:
         """
         Monitor main process and background items until done
         """
+
+        def wait_for_process(p):
+            """Wait for process and set main_code and self.quitting flag
+
+            Note that main_code is defined in the parent function,
+            so it is declared as nonlocal
+            """
+            nonlocal main_code
+            main_code = p.wait()
+            self.quitting = True
+
         self.write("Everything launched. Waiting to exit...")
         main_code = None
         span = self.span()
-        while not self.quitting and main_code is None:
+        Thread(target=wait_for_process, args=(main_process,))
```

Comment:
Can you clarify in a comment (or maybe the doc string above) why this is necessary? The intent of this change is clearly described in #1003, of course. But it would be even better to have a bit of context in the code itself. Perhaps something as simple as `# See https://github.com/telepresenceio/telepresence/issues/1003`.

Reply:
sure thing. added mention of 1003 in the docstring